flocker/node/agents/ebs.py

```diff
@@ -152,6 +201,62 @@ def compute_instance_id(self):
         """
         return get_instance_metadata()['instance-id'].decode("ascii")
 
+    def _get(self, blockdevice_id):
+        """
+        Lookup BlockDeviceVolume for given blockdevice_id.
+
+        :param unicode blockdevice_id: ID of a blockdevice that needs lookup.
+
+        :returns BlockDeviceVolume for the given input id.
+        """
+        for volume in self.list_volumes():
+            if volume.blockdevice_id == blockdevice_id:
+                return volume
+        raise UnknownVolume(blockdevice_id)
+
+    def _get_ebs_volume(self, blockdevice_id):
+        """
+        Lookup EBS Volume information for a given blockdevice_id.
+
+        :param unicode blockdevice_id: ID of a blockdevice that needs lookup.
+
+        :returns boto.ec2.volume.Volume for the input id. ``None`` if
+            no boto.ec2.volume.Volume was found for the given id.
+        """
+        for volume in self.connection.get_all_volumes():
+            if volume.id == blockdevice_id:
+                # Sync volume for uptodate metadata
```

Comment:
The volume just came back from a `get_all_volumes` call.  Does it already have out-of-date metadata?  Seems like it would be nice if boto gave us an object that was up-to-date at the time of the `get_all_volumes` call.